intel_extension_for_transformers/langchain/vectorstores/qdrant.py

```diff
@@ -0,0 +1,294 @@
+from intel_extension_for_transformers.transformers.utils.utility import LazyImport
+
+logging.basicConfig(
+    format="%(asctime)s %(name)s:%(levelname)s:%(message)s",
+    datefmt="%d-%M-%Y %H:%M:%S",
+    level=logging.INFO
+)
+
+if TYPE_CHECKING:
+    from qdrant_client.conversions import common_types
+
+_DEFAULT_PERSIST_DIR = './output'
+
+qdrant_client = LazyImport("qdrant_client")
+
+class Qdrant(Qdrant_origin):
+
+    _LANGCHAIN_DEFAULT_COLLECTION_NAME = "langchain"
+        
+    @classmethod
+    def from_documents(
+        cls,
+        documents: List[Document],
+        embedding: Embeddings,
+        sign: Optional[str] = None,
+        location: Optional[str] = None,
+        url: Optional[str] = None,
+        api_key: Optional[str] = None,
+        host: Optional[str]= None,
+        persist_directory: Optional[str] = None,
+        collection_name: Optional[str] = _LANGCHAIN_DEFAULT_COLLECTION_NAME,
+        force_recreate: Optional[bool] = False,
+        **kwargs: Any,
+    ):
+        """Create a Qdrant vectorstore from a list of documents.
+
+        Args:
+            documents (List[Document]): List of documents to add to the vectorstore.
+            embedding (Optional[Embeddings]): A subclass of `Embeddings`, responsible for text vectorization.
+            sign (Optional[str], optional): sign for retrieval_type of 'child_parent'. Defaults to None.
+            location (Optional[str], optional): 
+                If `:memory:` - use in-memory Qdrant instance.
+                If `str` - use it as a `url` parameter.
+                If `None` - fallback to relying on `host` and `port` parameters. 
+                Defaults to None.
+            url (Optional[str], optional): either host or str of "Optional[scheme], host, Optional[port],
+                Optional[prefix]". Defaults to None.
+            api_key (Optional[str], optional): API key for authentication in Qdrant Cloud. Defaults to None.
+            host (Optional[str], optional): Host name of Qdrant service. If url and host are None, set to
+                'localhost'. Defaults to None.
+            persist_directory (Optional[str], optional): Path in which the vectors will be stored while using
+                local mode. Defaults to None.
+            collection_name (Optional[str], optional): Name of the Qdrant collection to be used. 
+                Defaults to _LANGCHAIN_DEFAULT_COLLECTION_NAME.
+            force_recreate (bool, optional): _description_. Defaults to False.
+        """
+        if sum([param is not None for param in (location, url, host, persist_directory)]) == 0:
+            # One of 'location', 'url', 'host' or 'persist_directory' should be specified.
+            persist_directory = _DEFAULT_PERSIST_DIR
+            if sign == "child":
+                persist_directory = persist_directory + "_child"
+        assert not (sign == "child" and not persist_directory), \

```

Comment:
only check "persist_directory" is ok here. "sign" is an optional setting for child-parent retriever